sympy/physics/continuum_mechanics/beam.py

```diff
@@ -1416,97 +1416,74 @@ def plot_deflection(self, subs=None):
+            >>> b.apply_load(R1, 0, -1)
+            >>> b.apply_load(R2, 8, -1)
+            >>> b.apply_load(10000, 4, 0, end=8)
+            >>> b.bc_deflection = [(0, 0), (8, 0)]
+            >>> b.solve_for_reaction_loads(R1, R2)
+            >>> axes = b.plot_loading_results()
+        """
+        length = self.length
         variable = self.variable
         if subs is None:
             subs = {}
         for sym in self.deflection().atoms(Symbol):
             if sym == self.variable:
                 continue
             if sym not in subs:
-                raise ValueError('Value of %s was not passed.' % sym)
+                raise ValueError('Value of %s was not passed.' %sym)
         if self.length in subs:
             length = subs[self.length]
         else:
             length = self.length
 
-        # As we are using matplotlib directly in this method, we need to change
-        # SymPy methods to numpy functions.
-        shear = lambdify(variable,
-                         self.shear_force().subs(subs).rewrite(Piecewise),
-                         'numpy')
-        moment = lambdify(variable,
-                          self.bending_moment().subs(subs).rewrite(Piecewise),
-                          'numpy')
-        slope = lambdify(variable, self.slope().subs(subs).rewrite(Piecewise),
-                         'numpy')
-        deflection = lambdify(variable,
-                              self.deflection().subs(subs).rewrite(Piecewise),
-                              'numpy')
-
-        points = linspace(0, float(length), num=100*length)
-
-        # Creating a grid for subplots with 2 rows and 2 columns
-        fig, axs = plt.subplots(4, 1)
-        # axs is a 2D-numpy array containing axes
-        axs[0].plot(points, shear(points))
-        axs[0].set_title("Shear Force")
-        axs[1].plot(points, moment(points))
-        axs[1].set_title("Bending Moment")
-        axs[2].plot(points, slope(points))
-        axs[2].set_title("Slope")
-        axs[3].plot(points, deflection(points))
-        axs[3].set_title("Deflection")
-
-        fig.tight_layout()    # For better spacing between subplots
-        return axs
+        shear = self.shear_force().subs(subs).rewrite(Piecewise)
+        moment = self.bending_moment().subs(subs).rewrite(Piecewise)
+        slope = self.slope().subs(subs).rewrite(Piecewise)
+        deflection = self.deflection().subs(subs).rewrite(Piecewise)
+
+        ax1 = plot(shear, (variable, 0, length), title="Shear Force", xlabel='position', ylabel='Value', line_color='g', show=False)
+        ax2 = plot(moment, (variable, 0, length), title="Bending Moment", xlabel='position', ylabel='Value', line_color='b', show=False)
+        ax3 = plot(slope, (variable, 0, length), title="Slope", xlabel='position', ylabel='Value', line_color='m', show=False)
+        ax4 = plot(deflection, (variable, 0, length), title="Deflection", xlabel='position', ylabel='Value', line_color='r', show=False)
```

Comment:
I think the x and y labels should be the latex rendered variables used in the equations, probably x and y.

Reply:
The `x-label` could be made `xlabel='x'` (in latex of course)
But should the `y-label`  be made `ylabel='y'` (representing y-axis) or `ylabel="shear force"` or any other symbol for shear force such as `ylabel='V'`?

I guess we can use symbols like:
shear force = V
moment= M
slope= dy/dx
deflection = 𝛿

Should we use these symbols in `ylabel` of every plot?

Reply:
+1 for using symbols, they will take lesser space and hence make the plot neater. Also, name of the quantity in the y-axis will already be present in the title of the plot.

Reply:
I like your choice of variables for each plot.

Reply:
except slope. just use theta or it needs to be d Delta/dx